src/Umbraco.Core/Services/MetricsConsentService.cs

```diff
@@ -71,7 +71,7 @@ public void SetConsentLevel(TelemetryLevel telemetryLevel)
         IUser? currentUser = _backOfficeSecurityAccessor.BackOfficeSecurity?.CurrentUser;
         if (currentUser is null)
         {
-            currentUser = _userService.GetUserById(Constants.Security.SuperUserId);
+            currentUser = _userService.GetAsync(Constants.Security.SuperUserKey).GetAwaiter().GetResult();
```

Comment:
It would be really good to create a new async method here (`SetConsentLevelAsync`) and obsolete `SetConsentLevel`. `SetConsentLevel` should then call `SetConsentLevelAsync` with get await.

Reply:
Good point 👍